raiden/raiden_service.py

```diff
@@ -1082,7 +1087,10 @@ def mediated_transfer_async(
               expire.
         """
         if secret is None:
-            secret = random_secret()
+            if secret_hash is None:
+                secret = random_secret()
+            else:
+                secret = EMPTY_SECRET
 
         payment_status = self.start_mediated_transfer_with_secret(
```

Comment:
This has to be swapped around. `start_mediated_transfer_with_secret` is not the base case anymore, but the exception, now it should call `start_mediated_transfer_async` and not the other way around.

Reply:
Not sure I follow up here. Can you please explain what is wrong with current structure?

Reply:
Not a big problem, just that `start_mediated_transfer_with_secret` may not have a secret and the method name doesn't make much sense anymore

Reply:
OK. Thanks.